jdat_notebooks/preimaging/preimaging_03_association.ipynb

```diff
@@ -0,0 +1,454 @@
+    "association[\"code_version\"] = \"0.9.4\"  \n",
+    "association[\"target\"] = \"none\" \n",
+    "association[\"version_id\"] = \"null\" \n",
+    "association[\"degraded_status\"] = \"No known degraded exposures in association.\"\n",
+    "    \n",
+    "products_dict = {}\n",
+    "\n",
+    "products_dict[\"name\"] = \"lmc-\" + filtname \n",
+    "products_dict[\"members\"] = []   \n",
+    "for current_detector in bestlist:\n",
+    "    obs_dict = {\n",
+    "        \"expname\": current_detector,\n",
+    "        \"exptype\": \"science\"\n",
+    "    }\n",
+    "    products_dict[\"members\"].append(obs_dict) \n",
+    "\n",
+    "association[\"products\"] = [products_dict] "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# save association file\n",
+    "jsonfile = \"association-\" + filtname + \".json\"\n",
+    "with open(jsonfile, 'w') as json_file:\n",
+    "    json.dump(association, json_file, indent=4)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Execute the third stage of the NIRCam pipeline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here we execute the third stage of the NIRCam pipeline. In this stage the pipeline creates a mosaic of the input images and extracts a source catalogue."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Execute pipeline\n",
+    "# ----------------\n",
+    "\n",
+    "# run Image3 pipeline to get source catalog\n",
+    "im3 = Image3Pipeline(config_file='config/calwebb_image3.cfg')\n",
+    "#im3 = calwebb_image3.Image3Pipeline(config_file='calwebb_image3.cfg')\n",
+    "\n",
+    "# Options \n",
+    "# -------\n",
+    "\n",
+    "#im3.source_catalog.kernel_fwhm = # kernel_fwhm\n",
+    "#im3.source_catalog.kernel_xsize = # kernel_xsize\n",
+    "#im3.source_catalog.kernel_ysize = # kernel_ysize\n",
+    "#im3.source_catalog.npixels = # npixels\n",
+    "#im3.tweakreg_catalog.snr_threshold = 2000.0\n",
+    "#im3.resample.blendheaders = False\n",
+    "im3.tweakreg.skip = True\n",
+    "im3.tweakreg.enforce_user_order = True\n",
+    "im3.tweakreg.fitgeometry = 'shift' # valid values are 'shift', 'rscale', 'general'\n",
+    "im3.tweakreg.expand_refcat = True\n",
+    "im3.tweakreg.minobj = 5\n",
+    "im3.tweakreg.use2dhist = True\n",
+    "#im3.tweakreg.snr_threshold = 2000.0\n",
+    "im3.tweakreg.save_catalogs = True\n",
+    "#im3.tweakreg.xoffset = 7.5\n",
+    "#im3.tweakreg.yoffset = -7.2\n",
+    "im3.tweakreg.searchrad = 0.1 \n",
+    "im3.skymatch.skymethod = 'global' # valid values are: 'local', 'global', 'match', or 'global+match'\n",
+    "im3.source_catalog.skip = False \n",
+    "im3.output_file = 'lmc-f150w.fits'\n",
+    "\n",
+    "# Execute \n",
+    "# -------\n",
+    "\n",
+    "im3.run(jsonfile)"
```

Comment:
When using the run() method to call a pipeline, the configuration file is ignored and the pipeline is run with default settings (or whatever overrides you provide). https://jwst-pipeline.readthedocs.io/en/latest/jwst/stpipe/user_step.html#running-a-step-in-python